### ISSUE TITLE
Java 1.8 and script InstallJava3D.m

### ISSUE THREAD

**Issue by Tom-TBT** [fiji/fiji#141]
Hi,
I wanted to use the script InstallJava3D.m to use the 3Dviewer with Matlab, but it give me this error :
Undefined variable "ij3d" or class "ij3d.Install_J3D.getJava3DVersion" (line 46)

I had this error with Fiji bundled with java 1.8, so I tried it with the java 1.6, and it worked well.

Thank you,
Tom

**Comment by ctrueden** [comment]
Since you are using the new Fiji bundled with Java 8, you should not need to run this installer script anymore. Rather, Fiji itself ships with Java 3D 1.6 just like any other library. So the 3D Viewer should "just work" from MATLAB in this scenario (famous last words—I didn't try it). If not, please reply back with details of how it fails, and we can troubleshoot.

**Comment by Ajaxels** [comment]
Hi Curtis,
could you please specify how to use 3D viewer from Matlab without Miji? 
I was trying to follow the Matlab scripting page: [http://fiji.sc/MATLAB_Scripting] (http://fiji.sc/MATLAB_Scripting) the 'Running ImageJ within MATLAB' section, but after running the "ImageJ;" command got multiple errors starting with:

Error using ImageJ (line 42)
Java exception occurred: java.lang.UnsupportedClassVersionError: net/imagej/Main : Unsupported major.minor version 52.0

BR, Ilya

**Comment by ctrueden** [comment]
> Java exception occurred: java.lang.UnsupportedClassVersionError: net/imagej/Main : Unsupported major.minor version 52.0

Unfortunately, it seems that MATLAB (at least on OS X) ships with Java 7 instead of Java 8. And the current version of Fiji requires Java 8. So you may need to [do some surgery on your MATLAB](http://www.mathworks.com/matlabcentral/answers/103056-how-do-i-change-the-java-virtual-machine-jvm-that-matlab-is-using-for-mac-os) to switch it over to Java 8.

See also [this Troubleshooting topic](http://imagej.net/Troubleshooting#UnsupportedClassVersionError).

**Comment by Ajaxels** [comment]
Hi,
I updated the Java version of Matlab (on Windows) to 1.8.0_66-b18; after that I am starting 'ImageJ' in the Matlab command window. The Fiji interface is starting fine, but there are errors (see below).
The IMJ class is initialized in Matlab as well, but I can't send image from Matlab to Fiji. For example using the following command:
IJM.createImage('test', imread('snowflakes.png'), false);

Undefined function 'createImage' for input arguments of type 'net.imagej.matlab.ImageJMATLABCommands'.
 
----------------
[ERROR] Exception during event handling:
	[Event] org.scijava.ui.event.UIShownEvent
	context = org.scijava.Context@1625789b
	consumed = false
	ui = legacy
	[Subscriber] org.scijava.welcome.DefaultWelcomeService [priority = 0.0]
	[Method] protected void org.scijava.welcome.DefaultWelcomeService.onEvent(org.scijava.ui.event.UIShownEvent)
java.lang.RuntimeException: XPathFactory#newInstance() failed to create an XPathFactory for the default object model: http://java.sun.com/jaxp/xpath/dom with the XPathFactoryConfigurationException: javax.xml.xpath.XPathFactoryConfigurationException: java.util.ServiceConfigurationError: javax.xml.xpath.XPathFactory: jar:file:/D:/Matlab/R2015b/java/jarext/saxon9-xpath.jar!/META-INF/services/javax.xml.xpath.XPathFactory:2: Illegal configuration-file syntax
	at javax.xml.xpath.XPathFactory.newInstance(Unknown Source)

**Comment by hinerm** [comment]
@Ajaxels 

> IJM.createImage('test', imread('snowflakes.png'), false);

There are two different objects you can use for conversion.  I thought that method was a MIJ method... 

I will also try to create some better examples of going from MATLAB > ImageJ. The goal is that you can either call a utility method like "IJM.showImage(matlab array)", or just write scripts with a `@MatlabArray` parameter.

I probably won't be able to dig into this until next week. Thanks for the feedback though.

**Comment by Ajaxels** [comment]
Hi Mark,
thank you for the comment!
Upon starting ImageJ from Matlab the console window gives some small description, that also has the following text:
```
--- MIJ commands ---
For backwards compatibility, you can use MIJ to interact with ImageJ 1.x data structures. This is deprecated functionality.

``` 

So I was assuming that I can use IJM instead of MIJ... well I can do the following command:

`MIJ.createImage('test', imread('snowflakes.png'), true);`

but there are multiple errors again :(

Is there description of the IJM methods? Some small demo example would also be nice to have.
Thank you!

**Comment by hinerm** [comment]
>So I was assuming that I can use IJM instead of MIJ

Yeah that's a reasonable assumption! :smile: Ideally we either want to get rid of MIJ, or make IJM preserve the "MIJ.xxx" commands, as you expected. 

>Is there description of the IJM methods?

`IJM.help` or `IJM.usage` is supposed to print out the list of methods. Also the valid commands are supposed to be printed when you first run "ImageJ.m" from MATLAB.

>Some small demo example would also be nice to have.

Definitely! Thanks a lot for the feedback.

**Comment by hinerm** [comment]
@Ajaxels I finally had the time to revisit ImageJ-MATLAB. I fixed several usability issues and tried to make the IJM usage friendlier. I also added a parallel example for using [ImageJ in MATLAB](http://imagej.net/MATLAB_Scripting#Running_ImageJ_within_MATLAB). It's working well in my tests and I was able to run MATLAB with Java 8, although I didn't directly test 3D viewer.

Hope this helps. If you run into more problems or have any questions, let me know.
